src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1721,12 +1765,22 @@ else if (null == currentPrimaryPlaceHolder) {
             // Update timeout interval (but no more than the point where we're supposed to fail: timerExpire)
             attemptNumber++;
 
-            if (useParallel || useTnir) {
+            if (useParallel) {
                 intervalExpire = System.currentTimeMillis() + (timeoutUnitInterval * (attemptNumber + 1));
             }
             else if (isDBMirroring) {
                 intervalExpire = System.currentTimeMillis() + (timeoutUnitInterval * ((attemptNumber / 2) + 1));
             }
+            else if (useTnir) {
+                long timeSlice = timeoutUnitInterval * ((long) Math.pow(2, attemptNumber));
```

Comment:
It's a good idea to use shift operator. I changed to it base on your suggestion.

For the second comment: I had the same thought too. But our existing behaviour is 2,3,4 instead of 1,2,3,4, it starts at 2. Therefore, instead of 1,2,4, I thought I should keep the existing behaviour to make it start at 2 too, as 2,4,8.